### PR TITLE
ApiCompat CannotMakeMoreVisible rule shouldn't differentiate between Familiy and FamilyOrAssembly

### PR DESCRIPTION
**Pull request by ViktorHofer**
fixes https://github.com/dotnet/buildtools/issues/1418

### CODE REVIEW

File: src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public override DifferenceType Diff(IDifferences differences, ITypeDefinitionMem
             // If implementation is protected then contract must be protected as well. 
             if (impl.Visibility == TypeMemberVisibility.Family)
```
Comment (weshaggard) [inline src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs]: I think you also want to update the condition to add ` || impl.Visibility == TypeMemberVisibility.FamilyOrAssembly`.

File: src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public override DifferenceType Diff(IDifferences differences, ITypeDefinition im
             // If implementation is protected then contract must be protected as well. 
             if (impl.GetVisibility() == TypeMemberVisibility.Family)
             {
-                if (contract.GetVisibility() != TypeMemberVisibility.Family)
+                if (contract.GetVisibility() != TypeMemberVisibility.Family && contract.GetVisibility() != TypeMemberVisibility.FamilyOrAssembly)
```
Comment (weshaggard) [inline src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs]: I don't think types can really be `FamilyOrAssembly` (unless we go through this code path for nested types). At any rate it probably doesn't hurt anything to include it here. I would update the condition for the outter if as well like I called in the other comment.

**Comment by weshaggard** [comment]
@ViktorHofer thanks for fixing this. Did you verify that it removes the noise from the baseline files in corefx?

File: src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs (hunk 1 above)
Comment (ViktorHofer) [inline src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs]: Oooh yes, I only cared about from impl to contract but the other way should also be included. Will fix that.

**Comment by ViktorHofer** [comment]
Done. No haven't verified locally yet.

File: src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs [hunk 3]
```diff
@@ -7,56 +7,56 @@
...
-                        "Visibility of type '{0}' is '{1}' in the implementation but '{2}' in the contract.",
-                        impl.FullName(), impl.GetVisibility(), contract.GetVisibility());
-                    return DifferenceType.Changed;
-                }
-            }
-
-            return DifferenceType.Unknown;
-        }
-    }
+	// Removed because it appears the *MustExist rules already supercede these.
```
Comment (weshaggard) [inline src/ApiCompat/Rules/Compat/CannotMakeMoreVisible.cs]: spaces instead of tabs please.

**Comment by ViktorHofer** [comment]
@weshaggard Sorry for the formatting issue. Rebased and dropped last commit and push forced over it. Should be ok now

**Comment by weshaggard** [comment]
Thanks @ViktorHofer for the fix.

**Comment by ViktorHofer** [comment]
@weshaggard: Does it automatically get synced into corefx and if yes how long does it normally take? daily basis?

**Comment by weshaggard** [comment]
BuildTools doesn't automatically get updated, it is done manually on demand. I know @mellinoe is working on updating CoreFx's BuildTools to get his changes over https://github.com/dotnet/buildtools/pull/1422.

**Pull request merged**